internal/metamorph/processor.go

```diff
@@ -390,13 +410,10 @@ func (p *Processor) StartSendStatusUpdate() {
 					CompetingTxs: msg.CompetingTxs,
 				}
 
-				// if we receive new update check if we have client connection waiting for status and send it
-				p.responseProcessor.UpdateStatus(msg.Hash, StatusAndError{
-					Hash:         msg.Hash,
-					Status:       msg.Status,
-					Err:          msg.Err,
-					CompetingTxs: msg.CompetingTxs,
-				})
+				// if tx is seen on network or rejected, we don't expect any more status updates on this channel - remove from cache
+				if msg.Status == metamorph_api.Status_SEEN_ON_NETWORK || msg.Status == metamorph_api.Status_REJECTED {
+					p.delTxFromCache(msg.Hash)
```

Comment:
When the error is returned here, it should be logged but only if it is not of type `key not found in cache`